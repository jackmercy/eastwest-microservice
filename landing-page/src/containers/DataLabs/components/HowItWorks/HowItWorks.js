import { memo } from 'react';
import PropTypes from 'prop-types';

import { Content } from '../../../components';

import useStyles from './howItWorks.styles';

const HowItWorks = ({ title, description, data }) => {
  const classes = useStyles();

  return (
    <Content className={classes.content}>
      <h2 className={classes.title} data-test="how-it-works-title">
        {title}
      </h2>
      <p className={classes.description} data-test="how-it-works-description">
        {description}
      </p>
      <div className={classes.listItems} data-test="how-it-works-list-items">
        {data.map((item) => (
          <div
            key={item.key}
            className={classes.itemWrapper}
            data-test="how-it-works-item"
          >
            <div className={classes.itemInfoWrapper}>
              <img
                alt={item.title}
                src={item.icon}
                className={classes.itemIcon}
              />
              <div>
                <h2 className={classes.itemTitle}>{item.title}</h2>
                <p className={classes.itemDescription}>{item.description}</p>
              </div>
            </div>
            <img
              alt={item.title}
              src={item.image}
              className={classes.itemImage}
            />
          </div>
        ))}
      </div>
    </Content>
  );
};
HowItWorks.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  data: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      icon: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default memo(HowItWorks);
