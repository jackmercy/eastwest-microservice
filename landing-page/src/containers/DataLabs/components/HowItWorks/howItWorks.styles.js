import { createUseStyles } from 'react-jss';

import { rem } from '../../../../utils/jss';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  content: { paddingTop: rem(80), paddingBottom: rem(46) },
  title: {
    textAlign: 'center',
    fontWeight: 900,
    fontSize: rem(56),
    lineHeight: rem(72),
    margin: 0,
    transition: 'font-size .5s ease',
  },
  description: {
    textAlign: 'center',
    fontWeight: 600,
    fontSize: rem(24),
    lineHeight: rem(40),
    maxWidth: rem(1270),
    margin: [rem(24), 'auto', 0],
    transition: 'font-size .5s ease',
  },
  listItems: { marginTop: rem(120) },
  itemWrapper: {
    width: '100%',
    display: 'flex',
    justifyContent: 'space-between',
    '& + $itemWrapper': { marginTop: rem(130) },
  },
  itemImage: {
    flex: 'none',
    alignSelf: 'baseline',
    display: 'block',
    maxWidth: rem(430),
    maxHeight: rem(350),
    transition: 'max-width .3s ease',
  },
  itemInfoWrapper: { display: 'flex', marginRight: rem(16) },
  itemIcon: {
    flex: 'none',
    width: '100%',
    maxWidth: rem(64),
    marginRight: rem(36),
    alignSelf: 'baseline',
  },
  itemTitle: {
    fontWeight: 'bold',
    fontSize: rem(40),
    lineHeight: rem(64),
    margin: 0,
    textTransform: 'uppercase',
  },
  itemDescription: {
    width: '100%',
    maxWidth: rem(520),
    fontSize: rem(24),
    lineHeight: rem(32),
    marginTop: rem(16),
    marginBottom: 0,
    transition: 'max-width .3s ease',
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxMd}px)`]: {
    itemWrapper: { '& + $itemWrapper': { marginTop: rem(56) } },
    itemImage: { maxWidth: rem(300) },
    itemIcon: { maxWidth: rem(40), marginRight: rem(16) },
    itemTitle: { fontSize: rem(28), lineHeight: rem(32) },
    itemDescription: { fontSize: rem(20), lineHeight: rem(24) },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    content: { paddingTop: rem(40), paddingBottom: rem(40) },
    title: { fontSize: rem(32), lineHeight: rem(40) },
    description: { fontSize: rem(16), lineHeight: rem(24), marginTop: rem(16) },
    listItems: { marginTop: rem(40) },
    itemWrapper: {
      flexDirection: 'column-reverse',
      justifyContent: 'center',
      '& + $itemWrapper': { marginTop: rem(48) },
    },
    itemImage: { maxWidth: rem(180), alignSelf: 'center' },
    itemInfoWrapper: { marginTop: rem(24) },
    itemIcon: { maxWidth: rem(40) },
    itemTitle: { fontSize: rem(20), lineHeight: rem(32) },
    itemDescription: {
      marginTop: rem(8),
      fontSize: rem(14),
      lineHeight: rem(20),
    },
  },
});
