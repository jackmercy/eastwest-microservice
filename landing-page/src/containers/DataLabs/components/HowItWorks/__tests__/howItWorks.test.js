import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import HowItWorks from '../HowItWorks';

describe('Testing DataLabs components > HowItWorks', () => {
  let wrapper;
  const props = {
    title: 'How it works',
    description: 'Type something',
    data: [
      {
        key: '1',
        image: 'image',
        icon: 'icon',
        title: 'title',
        description: 'description',
      },
    ],
  };

  beforeAll(() => {
    wrapper = mount(<HowItWorks {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<HowItWorks {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a title', () => {
    expect(wrapper.find('[data-test="how-it-works-title"]')).toHaveLength(1);
  });

  it('should have a description', () => {
    expect(wrapper.find('[data-test="how-it-works-description"]')).toHaveLength(
      1,
    );
  });

  it('should show a list items ', () => {
    expect(wrapper.find('[data-test="how-it-works-list-items"]')).toHaveLength(
      1,
    );
    expect(wrapper.find('[data-test="how-it-works-item"]')).toHaveLength(
      props.data.length,
    );
  });
});
