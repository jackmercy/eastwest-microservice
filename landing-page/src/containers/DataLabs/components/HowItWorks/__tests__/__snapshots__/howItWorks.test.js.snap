// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DataLabs components > HowItWorks should render match snapshot 1`] = `
<Memo(Content)
  className="content-0-2-1"
>
  <h2
    className="title-0-2-2"
    data-test="how-it-works-title"
  >
    How it works
  </h2>
  <p
    className="description-0-2-3"
    data-test="how-it-works-description"
  >
    Type something
  </p>
  <div
    className="listItems-0-2-4"
    data-test="how-it-works-list-items"
  >
    <div
      className="itemWrapper-0-2-5"
      data-test="how-it-works-item"
      key="1"
    >
      <div
        className="itemInfoWrapper-0-2-7"
      >
        <img
          alt="title"
          className="itemIcon-0-2-8"
          src="icon"
        />
        <div>
          <h2
            className="itemTitle-0-2-9"
          >
            title
          </h2>
          <p
            className="itemDescription-0-2-10"
          >
            description
          </p>
        </div>
      </div>
      <img
        alt="title"
        className="itemImage-0-2-6"
        src="image"
      />
    </div>
  </div>
</Memo(Content)>
`;
